### PR TITLE
refactor Task.ParallelWithTrottle

### PR DESCRIPTION
**Pull request by sideeffffect**

### CODE REVIEW

**Comment by sideeffffect** [comment]
@tpetricek could you please have a look at this?

File: src/FSharpx.Extras/ComputationExpressions/Monad.fs [hunk 1]
```diff
@@ -1122,6 +1122,14 @@ module Task =
     /// Creates a task that runs the given task and ignores its result.
     let inline Ignore t = bind (fun _ -> returnM ()) t
 
+    /// Active pattern that matches on flattened inner exceptions in an AggregateException
+    let (|AggregateExn|_|) (e:exn) =
+        match e with
+        | :? AggregateException as ae -> ae.Flatten().InnerExceptions
+                                         |> List.ofSeq
+                                         |> Some
+        | _                           -> None
```
Comment (tpetricek) [inline src/FSharpx.Extras/ComputationExpressions/Monad.fs]: This is just a style preference, but I'm not a big fan of having the code in pattern matching aligned after `->`. I think it creates unnecessary indentation. So, I'd prefer something like:

    | :? AggregateException as ae -> 
        ae.Flatten().InnerExceptions |> List.ofSeq |> Some
    | _  -> None

Just a style thing though, so feel free to ignore this :-)

**Comment by tpetricek** [comment]
Looks good - if you want to change the indentation thing, you can, but don't feel like you have to :-).

BTW: If you plan on contributing to this project more actively, it feels like it would make sense to give you commit rights - I'm not that actively involved with it and you already did a bunch of nice additions! Happy to review things if you cc me, but I don't want to keep you waiting for new releases :)

**Comment by sideeffffect** [comment]
Done. Now we wait for @forki to review as well and approve, right?
I don't feel like I *need* commit rights, because I would be committing adhoc helper functions as I see the need for them. But I wouldn't mind helping up with other reviews and/or maintenance.
What would be really nice, is if somebody with admin privileges could enable AppVeyor in `Fsharpx._` and `ProjectScaffold` repositories. (Travis on Mono is already on, but this would be nice to have too.)
Thank you for the kind words!

**Comment by forki** [comment]
thx

**Comment by sideeffffect** [comment]
thanks for merging :)

**Pull request merged**